vertx-web/src/main/java/io/vertx/ext/web/Router.java

```diff
@@ -363,10 +363,24 @@ default void accept(HttpServerRequest request) {
    *
    * @param exceptionHandler  the exception handler
    * @return a reference to this, so the API can be used fluently
+   * @deprecated you should use {@link Router#errorHandler(int, Handler)} with 500 status code
    */
+  @Deprecated
   @Fluent
   Router exceptionHandler(@Nullable Handler<Throwable> exceptionHandler);
 
+  /**
+   * Specify an handler to handle an error for a particular status code. You can use to manage general errors too using status code 500.
+   * The handler will be called when the context fails and other failure handlers didn't write the reply or when an exception is thrown inside an handler.
+   * You <b>must not</b> use {@link RoutingContext#next()} inside the error handler
+   * This does not affect the normal failure routing logic.
+   *
+   * @param statusCode status code the errorHandler is capable of handle
+   * @param errorHandler error handler. Note: You <b>must not</b> use {@link RoutingContext#next()} inside the provided handler
+   * @return a reference to this, so the API can be used fluently
+   */
+  Router errorHandler(int statusCode, Handler<RoutingContext> errorHandler);
```

Comment:
shouldn't we use instead "failureHandler" ?

Reply:
as this is triggered by "fail"

Reply:
This method will handle both failures triggered by users (with `fail` methods), both internal router failures (like [query decoding error](https://github.com/vert-x3/vertx-web/pull/1142/files#diff-4f2099da2d67e8ac17bf1ad31e18f3c0R2373) ). I also think that `failureHandler` could be misleaded with Route's failure handlers... Maybe `globalFailureHandler` or something similar?

Reply:
Let's keep it error handler as we already have a handler called `ErrorHandler` that is crafted for this kind of cases, given a specific error, render a custom error message.